bitcoin/src/blockdata/block.rs

```diff
@@ -378,12 +378,23 @@ mod tests {
 
         assert_eq!(block.bip34_block_height(), Ok(100_000));
 
-        // block with 9-byte bip34 push
-        const BAD_HEX: &str = "0200000035ab154183570282ce9afc0b494c9fc6a3cfea05aa8c1add2ecc56490000000038ba3d78e4500a5a7570dbe61960398add4410d278b21cd9708e6d9743f374d544fc055227f1001c29c1ea3b0101000000010000000000000000000000000000000000000000000000000000000000000000ffffffff3d09a08601112233445566000427f1001c046a510100522cfabe6d6d0000000000000000000068692066726f6d20706f6f6c7365727665726aac1eeeed88ffffffff0100f2052a010000001976a914912e2b234f941f30b18afbb4fa46171214bf66c888ac00000000";
+        // block with 3-byte bip34 push for height 0x03010000 (non-minimal 1)
+        const BAD_HEX: &str = "0200000035ab154183570282ce9afc0b494c9fc6a3cfea05aa8c1add2ecc56490000000038ba3d78e4500a5a7570dbe61960398add4410d278b21cd9708e6d9743f374d544fc055227f1001c29c1ea3b0101000000010000000000000000000000000000000000000000000000000000000000000000ffffffff3703010000000427f1001c046a510100522cfabe6d6d0000000000000000000068692066726f6d20706f6f6c7365727665726aac1eeeed88ffffffff0100f2052a010000001976a914912e2b234f941f30b18afbb4fa46171214bf66c888ac00000000";
         let bad: Block = deserialize(&hex!(BAD_HEX)).unwrap();
 
-        let push = Vec::<u8>::from_hex("a08601112233445566").unwrap();
-        assert_eq!(bad.bip34_block_height(), Err(super::Bip34Error::UnexpectedPush(push)));
+        assert_eq!(bad.bip34_block_height(), Err(super::Bip34Error::NonMinimalPush));
+
+        // Block 15 on Testnet4 has height of 0x5f (15 PUSHNUM)
+        const BLOCK_HEX_SMALL_HEIGHT_15: &str = "000000200fd8c4c1e88f313b561b2724542ff9be1bc54a7dab8db8ef6359d48a00000000705bf9145e6d3c413702cc61f32e4e7bfe3117b1eb928071a59adcf75694a3fb07d83866ffff001dcf4c5e8401010000000001010000000000000000000000000000000000000000000000000000000000000000ffffffff095f00062f4077697a2fffffffff0200f2052a010000001976a9140a59837ccd4df25adc31cdad39be6a8d97557ed688ac0000000000000000266a24aa21a9ede2f61c3f71d1defd3fa999dfa36953755c690689799962b48bebd836974e8cf90120000000000000000000000000000000000000000000000000000000000000000000000000";
+        let block: Block = deserialize(&hex!(BLOCK_HEX_SMALL_HEIGHT_15)).unwrap();
+
+        assert_eq!(block.bip34_block_height(), Ok(15));
+
+        // Block 42 on Testnet4 has height of 0x012a (42)
```

Comment:
I guess I was thinking SMALL_HEIGHT as a test less than 16, Not sure if this test at 42 is needed really.

Reply:
The only test previously was a failure test for extremely long pushbytes.

This was to add the success case of a normal pushbytes.

I might add a few more to cover some more edge cases.

1. non-minimal PUSHDATA failure
2. non-minimal scriptNum encoding failure (DONE)
3. PUSHNUM success (DONE)
4. OP_FALSE / empty array / 0 success (DONE)
5. Overflow scriptNum failure

Still need 1 and 5, and as I mentioned in the other thread, collapsing 1 and 2 into the same Error variant might be confusing... (but then again, collapsing them might be confusing as well)

Reply:
Added tests for 1 and 5.

Reply:
> This was to add the success case of a normal pushbytes.

Then can I suggest a better name then `BLOCK_HEX_SMALL_HEIGHT_42` if it's for a normal case and not small block height.  Also I think hardcoding the blockheight number in the const should be dropped imo.